providers/flagd/pkg/provider.go

```diff
@@ -271,11 +429,81 @@ func (p *Provider) ObjectEvaluation(
 		}
 	}
 
-	return of.InterfaceResolutionDetail{
+	resDetail := of.InterfaceResolutionDetail{
 		Value: res.Value,
 		ProviderResolutionDetail: of.ProviderResolutionDetail{
 			Reason:  of.Reason(res.Reason),
 			Variant: res.Variant,
 		},
 	}
+
+	if p.isCacheAvailable() && res.Reason == flagdModels.StaticReason {
+		p.interfaceCache.Add(flagKey, resDetail)
+	}
+
+	return resDetail
+}
+
+func (p *Provider) isCacheAvailable() bool {
+	return p.cacheEnabled && p.Service.IsEventStreamAlive()
+}
+
+func (p *Provider) handleEvents(ctx context.Context) error {
+	eventChan := make(chan *schemaV1.EventStreamResponse)
+	errChan := make(chan error)
+
+	go p.Service.EventStream(ctx, eventChan, errChan)
+
+	for {
+		select {
+		case event, ok := <-eventChan:
+			if !ok {
+				if p.cacheEnabled { // disable cache
+					WithoutCache()(p)
+				}
+				return nil
+			}
+
+			switch event.Type {
+			case string(flagdService.ConfigurationChange):
+				if err := p.handleConfigurationChangeEvent(ctx, event); err != nil {
+					log.Errorf("handle configuration change event: %v", err)
+				}
+			}
+		case err := <-errChan:
+			if p.cacheEnabled { // disable cache
+				WithoutCache()(p)
+			}
+			return err
+		}
+	}
```

Comment:
What happens here in the case of a disconnection/re-connection? At least in the Java impl, the underlying gRPC libraries nicely handle re-connection. I assume that's the case here, but is it possible that we miss a flag change event while disconnected?

I think we may want to enhance flagd to send the latest flag config on connection, but a minimal solution might be to `Purge` the cache on reconnect. What do you think?

Reply:
Great shout, I've now implemented this as you suggested with a retry mechanism.